docs/storage/model-caching.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Model caching
+description: "Model caching allows you to quickly switch out machine learning models in your code."
+sidebar_position: 4
+---
+
+Model caching allows you to dynamically load and switch between machine learning models in your applications without rebuilding your contianer images or changing your code.
+It automatically handles model and dataset downloads and makes them available to your application.
+
+:::note
+
+Model caching currently supports models and data sets from [Hugging Face](https://huggingface.co/).
+
+:::
+
+## Benefits
+
+- Faster Development: Switch models instantly without rebuilding containers
+- Better Performance: Optimized cold start times and caching
+- Easy Integration: Works with popular ML frameworks like PyTorch and Transformers
+
+You can cache your models for both Pods and Serverless.
+
+## Get started with Serverless
+
+With model caching you can preload the models.
+This helps so you don't need to bake the model into your docker image or wait for the Worker to download your model from Hugging Face.
+
+1. Log in to the [RunPod Serverless console](https://www.runpod.io/console/serverless).
+2. Select **+ New Endpoint**.
+3. Provide the following:
+   1. Endpoint name.
+   2. Select your GPU configuration.
+   3. Configure the number of Workers.
+   4. (optional) Select **FlashBoot**.
+   5. (optional) Select a template.
+   6. Enter the name of your Docker image.
+      - For example `<username>/<repo>:<tag>`.
+   7. Specify enough memory for your Docker image.
+4. Add your Hugging Face model.
```

Comment:
It might be helpful to link users to examples at the bottom for clarity. We could also add step-by-step instructions or a screenshot on how to set the environment variables. This would make the process more user-friendly.